File: solutions/0283. Move Zeroes/0283.cpp
class Solution {
 public:
  void moveZeroes(vector<int>& nums) {
    int i = 0;
    for (const int num : nums)
      if (num != 0)
        nums[i++] = num;

    for (; i < nums.size(); ++i)
      nums[i] = 0;
  }
};

File: solutions/0283. Move Zeroes/0283.py
class Solution:
  def moveZeroes(self, nums: List[int]) -> None:
    j = 0
    for num in nums:
      if num != 0:
        nums[j] = num
        j += 1

    for i in range(j, len(nums)):
      nums[i] = 0
